### PR TITLE
[POS] Custom payment UI — Part 2 | Separating `CardReaderPaymentController`'s events from `MultiLiveEvent`

### PR DESCRIPTION
**Pull request by samiuelson**
### Separating CardReaderPaymentController's events from MultiLiveEvent

Closes: #12823

#### This is **part 2** of 5 PRs refactoring the Payment flow:
1. [[POS] Custom payment UI — Part 1 | Separating payment collection management from Android Framework](https://github.com/woocommerce/woocommerce-android/pull/12853)
2. **[[POS] Custom payment UI — Part 2 | Separating CardReaderPaymentController's events from MultiLiveEvent](https://github.com/woocommerce/woocommerce-android/pull/12870)**
3. [[POS] Custom Payment UI – Part 3 | Emitting UI–agnostic payment states in CardReaderPaymentController](https://github.com/woocommerce/woocommerce-android/pull/12877)
4. [[POS] Custom payment UI — Part 4 | Switch from viewStateData to paymentState](https://github.com/woocommerce/woocommerce-android/pull/12898)
5. [[POS] Custom payment UI — Part 5 | Unit tests clean up](https://github.com/woocommerce/woocommerce-android/pull/12943)

⚠️ Don't merge — the branch will be merged together with the above ones after additional testing of the whole refactor.

### Description

This PR refactors the way `CardReaderPaymentController` emits events:
1. `CardReaderPaymentController` emits now `sealed class CardReaderPaymentEvent` instances. This is safer to operate on a set of (sealed) classes bound to the parent class. Additionally, this removes the dependency on the `MultiLiveEvent.Event` and `MultiLiveEvent` in `CardReaderPaymentController`. These classes are not needed and desirable in Woo POS.
2. The existing `CardReaderPaymentViewModel` subscribes to the controller's `events` observable, maps it, and exposes in the form of the old model (extending `MultiLiveEvent.Event`) emitted by `LiveData`—retrofitting the existing UI implementation.

### Testing information

As a result of the refactoring done within this PR, the app should work without any change. It's crucial to test the IPP flow in the store management and POS modes against regression. It may be useful to base on the test plan (pdfdoF-5Jz-p2).

### The tests that have been performed

I tested the payment collection flow in both POS and store management modes, using card-present payment and TTP; and verified that the IPP flow works and is not changed. 
Configurations:
1. CA — payment collection/refund with Stripe test card (WisePad)
2. US — payment collection/refund with Stripe test card (TTP, M2)
3. Stripe and Woo Payments plugins

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>status: do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>CardReaderPaymentEvent</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt [hunk 1]
```diff
@@ -1628,7 +1629,7 @@ class CardReaderPaymentViewModelTest : BaseUnitTest() {
             whenever(cardReaderManager.collectPayment(any())).thenAnswer {
                 flow { emit(PaymentFailed(Generic, paymentData, "dummy msg")) }
             }
-
+            viewModel.event.observeForever {}
```
Comment (kidinov) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt]: np:
* Somewhere its `{}`, somewhere `{ }`
* Can we do it once in `initViewModel`?

**Review by kidinov — APPROVED** [review]
Code changes look good. Tests well too.

I left one np in the test, please take a look!

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt (hunk 1 above)
Comment (malinajirka) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt]: I second doing it once in init or similar => each test should IMO contain only lines that are affecting the result of the test for better readability => everything else should be done in an init block/@Before method/or similar.

Comment (samiuelson) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt]: Good idea – done in 4fc7a15

**Pull request merged**